sdk/src/test/scala/com/horizen/SidechainWalletTest.scala

```diff
@@ -375,6 +393,66 @@ class SidechainWalletTest
     sidechainWallet.scanPersistent(mockedBlock, withdrawalEpochNumber, feePaymentBoxes, Some(utxoMerkleTreeView))
   }
 
+  @Test
+  def testScanBackUp(): Unit = {
+    val mockedSecretStorage: SidechainSecretStorage = mock[SidechainSecretStorage]
+    val mockedWalletTransactionStorage: SidechainWalletTransactionStorage = mock[SidechainWalletTransactionStorage]
+    val mockedForgingBoxesInfoStorage: ForgingBoxesInfoStorage = mock[ForgingBoxesInfoStorage]
+    val mockedCswDataStorage: SidechainWalletCswDataStorage = mock[SidechainWalletCswDataStorage]
+    val mockedApplicationWallet: ApplicationWallet = mock[ApplicationWallet]
+    Mockito.when(mockedSecretStorage.getAll).thenAnswer(_=>secretList.toList)
+
+    //Create temporary WalletBoxStorage
+    val walletBoxStorageFile = temporaryFolder.newFolder("walletBoxStorage")
+    val walletBoxStorage = new SidechainWalletBoxStorage(new VersionedLevelDbStorageAdapter(walletBoxStorageFile), sidechainBoxesCompanion)
+
+    //Create temporary BackupStorage
+    val backupStorageFile = temporaryFolder.newFolder("backupStorage")
+    val backupStorage = new BackupStorage(new VersionedLevelDbStorageAdapter(backupStorageFile), sidechainBoxesCompanion)
+
+    boxToRestoreList.append(new Pair[ByteArrayWrapper, ByteArrayWrapper](new ByteArrayWrapper("key1".getBytes), new ByteArrayWrapper("value1".getBytes)))
+    backupStorage.update(getVersion, boxToRestoreList.asJava).get
+
+    val sidechainWallet = new SidechainWallet("seed".getBytes,
+      walletBoxStorage,
+      mockedSecretStorage,
+      mockedWalletTransactionStorage,
+      mockedForgingBoxesInfoStorage,
+      mockedCswDataStorage,
+      params,
+      mockedApplicationWallet)
+
+    sidechainWallet.scanBackUp(backupStorage.getIterator, sidechainBoxesCompanion)
+
+    assertTrue("Box stored to the backupStorage should be 7: 5 regular box + 1 new address + 1 fake ",boxToRestoreList.size == 7)
```

Comment:
I assume, that we should have only noncoin boxes. In case we meet coinbox -> than scanBackUp should return failure.
Check the test context. I think we need to test methods: successful backup with non-coin boxes, and failed cases with coinboxes.

Reply:
Done